ruamel.yaml-0.16.10 has change in handling empty keys -> TestYAMLResourceStore.test_empty_key

```
[  114s] =================================== FAILURES ===================================
[  114s] _____________________ TestYAMLResourceStore.test_empty_key _____________________
[  114s] 
[  114s] self = <translate.storage.test_yaml.TestYAMLResourceStore object at 0x7f000ded55b0>
[  114s] 
[  114s]         def test_empty_key(self):
[  114s]             store = self.StoreClass()
[  114s]             store.parse('''
[  114s]     '': Jedna
[  114s]     foo:
[  114s]       '': Dve
[  114s]     ''')
[  114s]             assert len(store.units) == 2
[  114s]             assert store.units[0].getid() == ''
[  114s]             assert store.units[0].source == 'Jedna'
[  114s]             assert store.units[1].getid() == 'foo->'
[  114s]             assert store.units[1].source == 'Dve'
[  114s]             out = BytesIO()
[  114s]             store.serialize(out)
[  114s] >           assert out.getvalue() == b'''? ''
[  114s]     : Jedna
[  114s]     foo:
[  114s]       ? ''
[  114s]       : Dve
[  114s]     '''
[  114s] E   assert b"'': Jedna\n...\n  '': Dve\n" == b"? ''\n: Jed...''\n  : Dve\n"
[  114s] E     At index 0 diff: b"'" != b'?'
[  114s] E     Full diff:
[  114s] E     - b"'': Jedna\nfoo:\n  '': Dve\n"
[  114s] E     + b"? ''\n: Jedna\nfoo:\n  ? ''\n  : Dve\n"
[  114s] E     ?   ++  ++                 ++  ++++
[  114s] 
```

--- a/requirements/optional.txt
+++ b/requirements/optional.txt
@@ -8,6 +8,6 @@ iniparse==0.4 ;python_version < '3.0'      # INI (ini2po)
 vobject==0.9.6.1     # iCal (ical2po)
 aeidon==1.2.1 ;python_version > '3.0'      # Subtitles (sub2po)
 l20n==4.0.0a1        # l20n (l20n2po)
-ruamel.yaml==0.16.5  # yaml
+ruamel.yaml==0.16.10 # yaml
 phply==1.2.5         # PHP
 cheroot==8.2.1       # tmserver

--- a/translate/storage/test_yaml.py
+++ b/translate/storage/test_yaml.py
@@ -6,8 +6,9 @@ import sys
 from io import BytesIO
 
 import pytest
+import ruamel.yaml
 
-from translate.storage import yaml, test_monolingual, base
+from translate.storage import base, test_monolingual, yaml
 
 
 class TestYAMLResourceUnit(test_monolingual.TestMonolingualUnit):
@@ -335,13 +336,14 @@ martin: {name: Martin D'vloper, job: Developer, skill: Elite}
   value: teststring2
 '''
 
+    @pytest.mark.skipif(ruamel.yaml.version_info < (0, 16, 6), reason='Empty keys serialization broken in ruamel.yaml<0.16.6')
     def test_empty_key(self):
-        store = self.StoreClass()
-        store.parse('''
-'': Jedna
+        yaml_souce = b''''': Jedna
 foo:
   '': Dve
-''')
+'''
+        store = self.StoreClass()
+        store.parse(yaml_souce)
         assert len(store.units) == 2
         assert store.units[0].getid() == ''
         assert store.units[0].source == 'Jedna'
@@ -349,12 +351,7 @@ foo:
         assert store.units[1].source == 'Dve'
         out = BytesIO()
         store.serialize(out)
-        assert out.getvalue() == b'''? ''
-: Jedna
-foo:
-  ? ''
-  : Dve
-'''
+        assert out.getvalue() == yaml_souce
 
     def test_dict_in_list(self):
         data = '''e1:

